test/unit/ResourceTypeRegistry.js

```diff
@@ -24,18 +24,47 @@ describe("ResourceTypeRegistry", function() {
     registry = new ResourceTypeRegistry();
   });
 
+  describe("constructor", () => {
+    it("should register resource descriptions provided in first parameter", () => {
+      let registry = new ResourceTypeRegistry([{
+        type: "someType",
+        info: "provided to constructor"
+      }]);
+      expect(registry.type("someType")).to.be.an.object;
+      expect(registry.type("someType").info).to.equal("provided to constructor");
+    });
+  });
+
   describe("type", () => {
-    let description = {
-      dbAdapter: {},
-      beforeSave: () => {},
-      beforeRender: () => {},
-      info: {},
-      urlTemplates: {"path": "test template"}
-    };
-
-    it("should be a getter/setter for a type",
-      makeGetterSetterTest(description, "mytypes", "type", true)
-    );
+    it("should merge descriptionDefaults into resource description", () => {
+      let registry = new ResourceTypeRegistry([], {
+        info: "provided as default"
+      });
+
+      registry.type("someType", {});
+      expect(registry.type("someType").info).to.equal("provided as default");
```

Comment:
This test had been [accidentally removed](https://github.com/andrewbranch/json-api/commit/e20d171f401c6c77241a6087cc731b4a322243e5) (whereas it was [this test](https://github.com/andrewbranch/json-api/commit/e20d171f401c6c77241a6087cc731b4a322243e5#diff-7d0a752b17e9bed5af29e8c6a2d707ebR37) that was inappropriately looking at the private functionality), so I added it back.